common/src/commonMain/kotlin/com/xacalet/minesweeper/common/ui/component/NumericDisplay.kt

```diff
@@ -22,35 +22,35 @@ sealed class DisplayValue {
 }
 
 @Composable
-fun SevenSegment(
+internal fun SevenSegment(
     modifier: Modifier = Modifier,
     displayValue: DisplayValue
 ) {
-    @DrawableRes val resource: Int = when (displayValue) {
-        DisplayValue.Minus -> R.drawable.ic_display_minus
+    val painter: Painter = when (displayValue) {
+        DisplayValue.Minus -> PainterResources.iconDisplayMinus()
         is DisplayValue.Number -> when (displayValue.number) {
-            0 -> R.drawable.ic_display_0
-            1 -> R.drawable.ic_display_1
-            2 -> R.drawable.ic_display_2
-            3 -> R.drawable.ic_display_3
-            4 -> R.drawable.ic_display_4
-            5 -> R.drawable.ic_display_5
-            6 -> R.drawable.ic_display_6
-            7 -> R.drawable.ic_display_7
-            8 -> R.drawable.ic_display_8
-            9 -> R.drawable.ic_display_9
+            0 -> PainterResources.iconDisplay0()
+            1 -> PainterResources.iconDisplay1()
+            2 -> PainterResources.iconDisplay2()
+            3 -> PainterResources.iconDisplay3()
+            4 -> PainterResources.iconDisplay4()
+            5 -> PainterResources.iconDisplay5()
+            6 -> PainterResources.iconDisplay6()
+            7 -> PainterResources.iconDisplay7()
+            8 -> PainterResources.iconDisplay8()
+            9 -> PainterResources.iconDisplay9()
             else -> throw IllegalArgumentException()
         }
     }
     Image(
-        modifier = modifier,
-        painter = painterResource(resource),
+        modifier = modifier.width(22.dp).height(42.dp),
```

Comment:
By doing so, you won't ever be able to resize the 7-segment display from the outside, as any size modification passed through the modifier will be override by the later `width` and `height` modifiers inside the composable. Instead, remove these modifiers and provide a `height` (width should be automatically set by the aspect ratio) through the modifier parameter.
```suggestion
        modifier = modifier,
```
However, I've noticed that after removing these modifiers, the original aspect ratio of the images wasn't respected. By setting `width` and `height` attributes, it looks that aspect ratio is kept again:
```xml
<svg
    xmlns="http://www.w3.org/2000/svg"
    width="29"
    height="55"
    viewBox="0 0 29 55"
    id="vector">
   ...
```